From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct cpuinfo_x86
Patch-mainline: never, kabi
References: kabi

In 4.4.117, commit ff891875c1a079c55cb5def525c2448c4cbed98c (x86/cpu:
Change type of x86_cache_size variable to unsigned int), upstream
commit 24dbc6000f4b9b0ef5a9daecb161f1907733765a changed an int to
unsigned int in struct cpuinfo_x86. It made the kABI checker to
complain.

Given the change is only in signedness and only low values are stored in
it, let's ignore this change and hide it from the kABI checker. Note
that the initial value during bootup is no longer negative. But nobody
should rely on that anyway.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/include/asm/processor.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/x86/include/asm/processor.h
+++ b/arch/x86/include/asm/processor.h
@@ -113,7 +113,11 @@ struct cpuinfo_x86 {
 	char			x86_vendor_id[16];
 	char			x86_model_id[64];
 	/* in KB - valid for CPUS which support this call: */
+#ifdef __GENKSYMS__
+	int			x86_cache_size;
+#else
 	unsigned int		x86_cache_size;
+#endif
 	int			x86_cache_alignment;	/* In bytes */
 	/* Cache QoS architectural values: */
 	int			x86_cache_max_rmid;	/* max index */
